### PR TITLE
Remove AsyncAlgorithms package

### PR DESCRIPTION
**Pull request by DePasqualeOrg**
Correct me if I'm wrong, but it seems that AsyncAlgorithms is not needed. LLMEval builds without it.

### CODE REVIEW

File: mlx-swift-examples.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -83,5 +65,5 @@
       }
     }
   ],
-  "version" : 2
+  "version" : 3
```
Comment (davidkoski) [inline mlx-swift-examples.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:68]: This update has caused problems for people in the past -- can you remove this line?

**Comment by davidkoski** [comment]
Yes, you are correct -- I think this is just leftovers from one of the early versions that had an async sequence as a generator type.

File: mlx-swift-examples.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (DePasqualeOrg) [reply mlx-swift-examples.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved]: Done.

**Review by davidkoski — APPROVED** [review]
Looks good, thank you!

**Pull request merged**